lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -1315,7 +1295,7 @@ def deploy_vm(self):
         self.configure_cpu_and_memory(vm_obj=vm_obj, vm_creation=True)
         self.configure_disks(vm_obj=vm_obj)
         self.configure_network(vm_obj=vm_obj)
-        if self.should_deploy_from_template() and len(self.params['customize']) > 0:
+        if vm_obj is None or self.should_deploy_from_template():
```

Comment:
this condition is correct as vm_obj is none on vm creation from scratch. This case is handled using same condition in every self.configure_xx
why customization should be forbidden ? If i want to create a VM from scratch using a PXE boot and after your sysprep ?